### ISSUE TITLE
Bug: Porkbun new wildcard entry instead of updating existing one

### ISSUE THREAD

**Issue by Korunos47** [qdm12/ddns-updater#731]
New A wildcard entry on porkbun instead of updating the existing one

1. Is this urgent: No
2. DNS provider(s) you use: porkbun
3. Program version:

Running version latest built on 2024-05-23T13:33:03.304Z (commit 8e09cd6)

4. What are you using to run the container: unraid app (CA)
5. Extra information (optional)

Logs:

```
INFO reading JSON config from file /updater/data/config.json
INFO Found 3 settings to update records
INFO Reading history from database: domain domain.com host palworld ipv4
INFO Reading history from database: domain domain.com host @ ipv4
INFO Reading history from database: domain domain.com host * ipv4
INFO [healthcheck server] listening on 127.0.0.1:9999
INFO [backup] disabled
INFO [http server] listening on :8000
INFO obtaining ipv4 address succeeded after 1 failed try
INFO ipv4 address of any.domain.com is XXX.XXX.XXX.27 and your ipv4 address  is XXX.XXX.XXX.151
INFO Updating record [domain: domain.com| host: * | provider: porkbun | ip: ipv4] to use XXX.XXX.XXX.151
ERROR creating record: HTTP status is not valid: 400: Create error: We were unable to create the DNS record.

```

Configuration file (**remove your credentials!**):

```{
  "settings": [
    {
      "provider": "porkbun",
      "domain": "domain.com
      "host": "palworld,@,*",
      "api_key": "KEY",
      "secret_api_key": "KEY",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}

```

Host OS:
Unraid

**Comment by TheGreatestJannet** [comment]
Same issue here. I have had to manually remove all the wildcard entries or it seems stuck on 'We were unable to create the DNS record'.

**Comment by Korunos47** [comment]
Exactly. I noticed an increase of loading times for my container until I found that I had 5 different dns entries for my wildcard. After removing them it went back to normal.

**Comment by jawaff** [comment]
Same problem on my end as well. Everything still works, but it almost defeats the purpose of this project.

**Comment by videah** [comment]
Running into this issue as well, seems to be an old one going by [this issue](https://github.com/qdm12/ddns-updater/issues/247).

**Comment by qdm12** [comment]
Can you try pulling the image? I pushed 987138dfc165f3b087d494cf4c3ed605bbeba529 which may fix this.
TLDR: before it was getting all records with `https://porkbun.com/api/json/v3/dns/retrieveByNameType/domain.com/A/*` but the `*` at the end is not really a valid URL character, so now it's removed (for both `*` and `@`, not just `@`). That could had explained it would re-create it because it wouldn't find it in the first place.

**Comment by Korunos47** [comment]
Seems okay now. Will test a few more days and report again.

**Comment by qdm12** [comment]
Still good? About to make a v2.6.0 release. You can try by changing your A record to 127.0.0.1 on the porkbun webui and let ddns-updater detect the required update, to force a test.

**Comment by Korunos47** [comment]
Thats what I did basically. It's working now. Thank you!

**Comment by qdm12** [comment]
Awesome, thanks for getting back. Closing this! 👍
